src/Controller/CRUDController.php

```diff
@@ -285,6 +289,7 @@ public function editAction(Request $request): Response
 
             // persist if the form was valid and if in preview mode the preview was approved
             if ($isFormValid && (!$this->isInPreviewMode() || $this->isPreviewApproved())) {
+                /** @phpstan-var T $submittedObject */
```

Comment:
I had no error on 3.x ; but if it's needed we could add it to 3.x
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Controller/CRUDController.php#L332

Reply:
I don't know in `3.x`, I noticed that failed by the time I was done merging and there were some conflicts, so I'd rather add it also in `3.x` but not blocking this one.

Reply:
Sure it's not a blocking. I would say the only blocking point is 
```
 public function apply(ProxyQueryInterface $query, array $filterData): void;
```
Since we've tried to add all the possible typehint in master ; let's try to keep it ; if it's okay for you.

Reply:
sure, the thing is that there was `mixed` in phpdoc, I added `ProxyQueryInterface` but I missed that one.